src/lib/runtime/Web3FunctionRunner.ts

```diff
@@ -267,9 +285,18 @@ export class Web3FunctionRunner {
           memory,
           rpcCalls,
           network: networkStats,
+          throttled,
         };
       }
     } else {
+      if (error && error.message) {
+        throttled.memory = error.message.includes("Memory limit exceeded");
+        throttled.rpcRequest = error.message.includes(
+          "RPC requests limit exceeded"
+        );
+        throttled.duration = error.message.includes("exceed execution timeout");
```

Comment:
I don't like much the `message.includes` detection.
If we update the message later, we will break this functionality without noticing it.

Maybe instead we should customize the error thrown by the runtime/sandbox when they detect the type of error.
Like adding a `error.throttledType = "memory" | "duration" | "rpcRequests"` and then testing this with a stricter typing validation